ADR/0027-availability-probe-framework.md

```diff
@@ -0,0 +1,125 @@
+# 27. Availability Probe Framework
+
+* Date 2023-07-06
+
+## Status
+
+Accepted
+
+## Context
+
+As an AppStudio developer building functionality for the platform, I want to be able to
+easily visualize and comprehend the stability and availability of deployed systems in
+order to inform and influence future work towards improving the overall system
+reliability.
+  
+Such indication should tell us the overall up time of AppStudio with respect to services
+under the control of AppStudio developers.
+
+AppStudio is defined to be available at a given moment if all of its components are
+reporting to be available at that given moment.
+
+A component is defined to be available at a given moment if all of its availability
+probes are reporting to be available at that given moment.
+
+A convention is required for providing the availability of a probe.
+
+Once this is in place, those indicators can be aggregated in order to report the overall
+availability of AppStudio.
+
+## Decision
+
+Probes' availability will be provided as a Prometheus metric. The metric will contain
+metric labels to allow differentiating between the different probes. When exported out
+of the cluster of origin, additional labels will be attached to the metric to mark the
+cluster in which the metric was generated.
+
+### Delivery Method
```

Comment:
What do you mean by delivery method? The delivery of the availability probe? By following a standard pattern which will enable automatic ingestion into a higher level availability? This section might be better named something else as I don't see any clarification of this term in the first couple paragraphs of the section.

Reply:
I renamed the section.